### DISCUSSION TITLE
How do I add custom mapping to telescope?

### DISCUSSION THREAD

**Discussion by idelice** [LazyVim/LazyVim#3106] [General]
### Scenario

I open up telescope find files, and search for a file called `foo`, but I only want the file type `java`. 

So normally I would do: `foo .java$` and it works fine, but I want to add a custom mapping that prefixes the file type e.g. it could be `<c-t>` that would prefix with: `<space> .java`.

Is this possible and if so; where do I add this mapping?

**Comment by heshify** [comment]
You can add this code into your keys = {} inside your telescope.lua file.

```lua 
{
      "<C-t>",
      function()
        require("telescope.builtin").find_files({
          prompt_title = "Java Files",
          find_command = { "rg", "--files", "--type", "java" },
        })
      end,
      desc = "Find Java Files",
}
```
